src/Bicep.Core/Semantics/ArmTemplateSemanticModel.cs

```diff
@@ -66,7 +66,8 @@ public ArmTemplateSemanticModel(ArmTemplateFile sourceFile)
                         GetType(parameterProperty.Value),
                         parameterProperty.Value.DefaultValue is null
                             ? TypePropertyFlags.WriteOnly | TypePropertyFlags.Required
-                            : TypePropertyFlags.WriteOnly))
+                            : TypePropertyFlags.WriteOnly,
+                        parameterProperty.Value.Metadata?.Value?.SelectToken("description")?.ToString()))
```

Comment:
[nit] Technically this field could contain anything, so I think it would be worth creating a helper method to read the description from metadata, and explicitly checking that that JToken is a string.